rosutil/convenience.rkt

```diff
@@ -8,6 +8,9 @@
   [fresh-symbolic (-> (or/c symbol? string?)
                       @solvable?
                       @constant?)]
+  [fresh-memory-like (-> (or/c symbol? string?)
+                         vector?
+                         (vectorof @constant?))]
```

Comment:
This `vectorof` will involve every element of the vector being checked when this function returns. If we expect `fresh-memory-like` to be called in hot code, it might be worth exporting a version of this that is not protected by a contract. I don't know if there's a standard way to do this sort of thing, but I've adopted the convention of exporting the same indentifier but without a contract as `x/unchecked`, e.g. https://github.com/anishathalye/rtl/blob/master/rosutil/dependence.rkt#L8

Reply:
Good to know, I added the extra unchecked export!

Right now the only place my code consumes this function is indirectly through the implementation of the overapproximate hint, and it didn't seem to make much of a noticeable difference (I overapproximate a memory each cycle, but perhaps it's not big enough to be much of an issue). 

I'll keep this in mind in case this does become a problem, and modify the overapproximate implementation with the unchecked version if needed.